src/other/Persist.js

```diff
@@ -31,7 +31,144 @@
             }
             return retVal;
         },
-
+        serializeTheme: function(widget,filter){
+            return JSON.stringify(this.serializeThemeToObject(widget,filter));
+        },
+        serializeThemeToObject: function (widget,filter){
+            var context = this;
+            
+            var propObj = {};
+            _paramCrawler(widget,filter);
+            return propObj;
+            
+            function _paramCrawler(widget,filter){
+                var retArr = [];
+                filter = typeof(filter) === "undefined" ? ["surface","Color","Font","palette"] : filter;
+                context.discover(widget).forEach(function (item) {
+                    if (widget[item.id + "_modified"]() || typeof(widget["__meta_"+item.id].trueDefaultValue) !== "undefined") {
+                        switch (item.type) {
+                            case "widget":
+                                var tempArr = _paramCrawler(widget[item.id](),filter);
+                                retArr = retArr.concat(tempArr);
+                                break;
+                            case "widgetArray":
+                                var widgetArray = widget[item.id]();
+                                widgetArray.forEach(function (widget) {
+                                    var tempArr = _paramCrawler(widget,filter);
+                                    retArr = retArr.concat(tempArr);
+                                }, this);
+                                break;
+                            default:
+                                if(_isFilterMatch(item.id,filter)){
+                                    var paramIsIncluded = false;
+                                    var classParts = widget._class.trim().split(" ");
+                                    classParts.forEach(function(c) { 
+                                        if(!paramIsIncluded){
+                                            if(propObj[c] === undefined){
+                                                propObj[c] = {};
+                                            } 
+                                            if (propObj[c][item.id] === undefined) { 
+                                                propObj[c][item.id] = widget[item.id](); 
+                                                paramIsIncluded = true;
```

Comment:
I've switched them to **for** just now. Array.some sounds interesting, I havent used it before.